infrastructure/src/main/java/org/corfudb/infrastructure/ManagementAgent.java

```diff
@@ -800,6 +833,10 @@ public void shutdown() {
         try {
             initializationTaskThread.interrupt();
             initializationTaskThread.join(ServerContext.SHUTDOWN_TIMER.toMillis());
+        } catch (InterruptedException ie) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either re-interrupt this method or rethrow the "InterruptedException". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2142)